drivers/edac/edac_nxp.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/edac.h>
+#include <zephyr/drivers/edac/edac_nxp.h>
+#include <fsl_erm.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(edac_nxp, CONFIG_EDAC_LOG_LEVEL);
+
+#ifdef CONFIG_EDAC_ERROR_INJECT
+#include <fsl_eim.h>
+
+#define EDAC_NXP_SINGLE_BIT_ERROR_MASK 0x1
+#define EDAC_NXP_DOUBLE_BIT_ERROR_MASK 0x3
+
+struct edac_nxp_eim_channel {
+	uint32_t start_address;
+	uint32_t size;
+	uint32_t inject_enable;
+	uint32_t ecc_enable;
+	uint8_t channel_id;
+};
+
+/* NOTE to checkpatch error: SPACING: need consistent spacing around '*' (ctx:WxV)  */
+#define EIM_CONFIG                                                                                 \
+	EIM_Type *eim_base;                                                                        \
+	const struct edac_nxp_eim_channel *eim_channels;                                           \
+	uint8_t eim_channel_num;
+#else
+#define EIM_CONFIG
+#endif /* CONFIG_EDAC_ERROR_INJECT */
```

Comment:
this doesn't need to be a macro, just #ifdef within the config struct definition, its actually less confusing that way

Reply:
@keith-zephyr I think the compliance script is doing something wrong here, I saw you are coding guideline maintainer, can you help figuring out what is the problem here? Because this code in the PR now is weird due to trying to get around this script, I guess